scripts/installer.nsi

```diff
@@ -285,6 +285,19 @@ Section "XMake (required)" InstallExeutable
 
 SectionEnd
 
+Section "Enable Long Path" LongPath
+  ; Get git path from the registry
+  ReadRegStr $R0 ${HKLM} "SOFTWARE\GitForWindows" "InstallPath"
+  ${If} $NOADMIN == "false"
+    ; In admin enable long path on both git system config and in registry
+    WriteRegDWORD ${HKLM} "SYSTEM\CurrentControlSet\Control\FileSystem" "LongPathsEnabled" 1
+    Exec '"$R0\cmd\git.exe" config --system core.longpaths true'
```

Comment:
If git is not installed on the user's system, or if the installation path is not cmd\git.exe, the installation may fail.

And, xmake may install git in its own package installation directory.

Reply:
Only for installing the xmake-repo package does git need to support long paths, but for some packages that need to handle long paths, we will turn on long paths to support it. 

https://github.com/xmake-io/xmake-repo/blob/e1581652ef8cdfb3605d341eebea240a05a580f9/packages/l/libtorch/xmake.lua#L45

So I don't think git's setting of longpath is necessary, it's automatically supported by xmake when installing packages.

Reply:
Also, this setting may have an effect on other programs and it is recommended that more detailed descriptive information be provided to tell the user the context in which it is used.

Reply:
I didn't know about that policy, so I will remove the execution of git. The description will be modified to inform the user that it needs to reboot to make sure that all program have registered the new key value. 
From the Microsoft docs:
> The registry key will not be reloaded during the lifetime of the process. In order for all apps on the system to recognize the value of the key, a reboot might be required because some processes may have started before the key was set.